### DISCUSSION TITLE
Improved Component Browser

### DISCUSSION THREAD

**Discussion by farmaazon** [enso-org/enso#6068] [Show and tell]
Due to @vitvakatu recent weeks of hard work, we recently merged important improvements of Component Browser User Experience.

Foremost, the suggestions are inserted at the cursor position (before, they were always added at the end). Additionally, they replace the currently edited name only (not entire chunks of code as before). Now, we can use suggestions when constructing arrays:

https://user-images.githubusercontent.com/3919101/227482897-dd12213b-729a-4efb-b02b-37a12b71693a.mp4

And we can now just edit expressions in the middle of the nodes (here we replace `First` with `Sum`).

https://user-images.githubusercontent.com/3919101/227483333-458daf70-3683-4393-8ca3-3717be98f706.mp4

Second, when typing literals, a new special suggestion for literal appears. Now you do not need to remember to press cmd+enter after typing literal. Moreover, the closing quotes will be added automatically.

https://user-images.githubusercontent.com/3919101/227484027-8d39f851-6b69-48b0-a5de-4e8248765a82.mp4

There are still known issues thought:
1. Due to [the bug in the parser](#6041), the expressions like `[Aggregate_Column` are treated as a single edited name, somewhat fooling our filtering. To mitigate it, you may put space after opening bracket, or add closing bracket and move back.
2. The current suggestion is still always accepted on enter, what may still give unexpected behavior when, for example, just typed closing bracket (as brackets are not literals).
3. The lack of focus may be more irritating.

Anyway, have a fun with new Component Browser!

**Comment by wdanilo** [comment]
This is so freaking amazing @vitvakatu and @farmaazon!!! ❤️
